Makefile

```diff
@@ -2,7 +2,11 @@
 default: build
 NAME:=karina
 
-VERSION_TAG=$(VERSION)-$(shell date +"%Y%m%d%H%M%S")
+ifeq ($(VERSION),)
+  VERSION_TAG=$(shell git describe --abbrev=0 --tags)-$(shell date +"%Y%m%d%H%M%S")
```

Comment:
For sanity when handling multiple local builds could I suggest:
```suggestion
  VERSION_TAG=$(shell git describe --abbrev=0 --tags --exact-match 2&>/dev/null || git describe --abbrev=0 --all)-$(shell date +"%Y%m%d%H%M%S")
```
this will set VERSION to the name of the branch being built in when the commit isn't exactly release-tagged.

@moshloop I see there's also main.date and main.commit vars, would it make more sense to set main.date ldflag rather than putting a date string in the version information?

Reply:
@BrendanGalloway Thanks for the suggestion, I committed the same for now :)

Reply:
@BrendanGalloway I tried your suggestion and unfortunately that again skips the release tags, reference the Image below. 
![image](https://user-images.githubusercontent.com/36641301/123438817-b628ec00-d5ee-11eb-93eb-f8d0db2bda25.png)

Maybe we can use "--always" flag as the fallback when tags are not available. 
![image](https://user-images.githubusercontent.com/36641301/123438944-dbb5f580-d5ee-11eb-96ff-c5686f5886de.png)

Reply:
@apaarshrm39  - the idea is less that tags would not be available, but more that when working in local repos the most recent tag in history will be picked up, which would result in dev builds having (incorrect) release version info

It looks like you accidentally inserted a space between `&` and `>` in the version you tested (you've amended it out so I can't be sure):

```bash
~/Flanksource/karina(testenv)$ sed -n 6p Makefile 
        VERSION_TAG=$(shell git describe --abbrev=0 --tags --exact-match 2& >/dev/null || git describe --abbrev=0 --all)-$(shell date +"%Y%m%d%H%M%S")
~/Flanksource/karina(testenv)$ make
fatal: Not a valid object name 2
go build -o ./.bin/karina -ldflags "-X \"main.version=-20210625170658\""  main.go
~/Flanksource/karina(testenv)$ #Fix space in makefile
~/Flanksource/karina(testenv)$ sed -n 6p Makefile 
        VERSION_TAG=$(shell git describe --abbrev=0 --tags --exact-match 2&>/dev/null || git describe --abbrev=0 --all)-$(shell date +"%Y%m%d%H%M%S")
~/Flanksource/karina(testenv)$ make
go build -o ./.bin/karina -ldflags "-X \"main.version=heads/testenv-20210625170711\""  main.go
```
the `invalid object name 2` indicates 2 was passed to git as an argument, rather than used as a redirect for stderr

Reply:
@BrendanGalloway Thanks, I don't believe  there was any spaces as I just committed your suggestion. Plus I rechecked it again.

 It was something with 2&>/dev/null itself.

To test this I tried to redirect the output to a file instead of /dev/null and see : 
![image](https://user-images.githubusercontent.com/36641301/123462695-3f005180-d608-11eb-9da8-bf65cabb1105.png)

The "2&>" command is busted anyway, it's that the shell used by the makefile doesn't suppress errors
One alternative to make the above command work anyway is to change the shell in makefile to /bin/bash
but,
also I couldn't understand the reason for using 2&>/dev/null as 
1) 2 is used to redirect STDERR
2)& is used to redirect both STDERR and STDOUT  

it seems to be sending the STDOUDT to null to 
If we are just trying to send the error to /dev/null so I believe just 2 alone should do the trick.

Reply:
@apaarshrm39

You are correct, it should just be `2>` not `2&>`.  I see you sent that up in 88b1505 - does that work better on your side, or does it still print the error about 2 not being a valid object?

Reply:
@BrendanGalloway, Yeah it work as intended on my side now ! Thank you !